x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_rules_client.ts

```diff
@@ -87,22 +84,30 @@ export class RuleMigrationsDataRulesClient extends RuleMigrationsDataBaseClient
   }
 
   /** Updates an array of rule migrations to be processed */
-  async update(ruleMigrations: UpdateRuleMigrationInput[]): Promise<void> {
+  async update(ruleMigrations: UpdateRuleMigrationData[]): Promise<void> {
     const index = await this.getIndexName();
 
-    let ruleMigrationsSlice: UpdateRuleMigrationInput[];
+    let ruleMigrationsSlice: UpdateRuleMigrationData[];
     const updatedAt = new Date().toISOString();
     while ((ruleMigrationsSlice = ruleMigrations.splice(0, BULK_MAX_SIZE)).length) {
       await this.esClient
         .bulk({
           refresh: 'wait_for',
           operations: ruleMigrationsSlice.flatMap((ruleMigration) => {
-            const { id, ...rest } = ruleMigration;
+            const {
+              id,
+              translation_result: translationResult,
+              elastic_rule: elasticRule,
+              ...rest
+            } = ruleMigration;
             return [
               { update: { _index: index, _id: id } },
               {
                 doc: {
                   ...rest,
+                  elastic_rule: elasticRule,
+                  translation_result:
+                    translationResult ?? convertEsqlQueryToTranslationResult(elasticRule?.query),
```

Comment:
Here we update translation result based on the query. If query is not passed for update, we will not update `translation_result`. Also, `translation_result` will be used if passed as in `UpdateRuleMigrationData` object.